### PR TITLE
Support runtime config in the spintainer executor

### LINKED ISSUE #319: Ensure runtime config works with spintainer executor
Also write an e2e test to cover this case.

### PR DESCRIPTION
**Pull request by calebschoepp**
Closes #319.

I tested locally to make sure that this works both when runtime config is specified and when it isn't.

### CODE REVIEW

**Comment by endocrimes** [comment]
Could you make sure this is covered by tests at the same time?

(I'm a bit worried about spintainer feature drift because I assume it'll be used much less frequently, covering in tests for the big things should help us there 😅)

**Comment by calebschoepp** [comment]
@endocrimes I've added some tests.

File: e2e/redis_test.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+				t.Fatal(err)
+			}
+			return ctx
+		}).
+		Assess("spin app deployment and service are available", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// wait for deployment to be ready
+			if err := wait.For(
+				conditions.New(client.Resources()).DeploymentAvailable(testSpinAppName, testNamespace),
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(30*time.Second),
```
Comment (endocrimes) [inline e2e/redis_test.go:82]:
```suggestion
				wait.WithInterval(time.Second),
```

30 seconds between checks is a _really_ long time (especially as the first check doesn't happen until after that interval) - and the check is ~free, so cut this down.

File: e2e/redis_test.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+			svc := &v1.ServiceList{
+				Items: []v1.Service{
+					{ObjectMeta: metav1.ObjectMeta{Name: testSpinAppName, Namespace: testNamespace}},
+				},
+			}
+
+			if err := wait.For(
+				conditions.New(client.Resources()).ResourcesFound(svc),
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(30*time.Second),
```
Comment (endocrimes) [inline e2e/redis_test.go:96]:
```suggestion
				wait.WithInterval(500*time.Millisecond),
```

this should already be true before the check above completes, so make this be much faster

File: e2e/redis_test.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
```
Comment (endocrimes) [inline e2e/redis_test.go:217]:
```suggestion
				wait.WithInterval(500*time.Millisecond),
```

as above

**Review by endocrimes — CHANGES_REQUESTED** [review]
Some blocking changes in the tests (timeouts and validity of a couple of the checks) otherwise LGTM

**Comment by calebschoepp** [comment]
Nice catch @endocrimes, I made those fixes to the spintainer e2e too and overall this shaved 4 minutes off the e2e tests.

**Review by endocrimes — APPROVED** [review]
✨

**Pull request merged**